tests/sources/test_redis.py

```diff
@@ -335,3 +331,47 @@ async def test_advanced_rules_validation(advanced_rules, expected_validation_res
             advanced_rules
         )
         assert validation_result == expected_validation_result
+
+
+@pytest.mark.asyncio
+async def test_client_when_mutual_ssl_enabled():
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.client.ssl_enabled = True
+        source.client.mutual_tls_enabled = True
+        source.client.store_ssl_key = Mock(return_value="/tmp/tmp4vn0jxy7.crt")
+        with mock.patch(
+            "redis.asyncio.from_url",
+            return_value=AsyncMock(),
+        ) as from_url_mock:
+            _ = source.client._client
+            connection_string = "rediss://username:password@localhost:6379?ssl_certfile=/tmp/tmp4vn0jxy7.crt&ssl_keyfile=/tmp/tmp4vn0jxy7.crt"
+            from_url_mock.assert_called_with(connection_string, decode_responses=True)
+
+
+@pytest.mark.asyncio
+async def test_client_when_ssl_enabled():
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.client.ssl_enabled = True
+        with mock.patch(
+            "redis.asyncio.from_url",
+            return_value=AsyncMock(),
+        ) as from_url_mock:
+            _ = source.client._client
+            connection_string = "rediss://username:password@localhost:6379"
+            from_url_mock.assert_called_with(connection_string, decode_responses=True)
+
+
+@pytest.mark.asyncio
+async def test_ping_when_mutual_ssl_enabled():
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.client.ssl_enabled = True
+        source.client.mutual_tls_enabled = True
+        with mock.patch(
+            "redis.asyncio.from_url",
+            return_value=AsyncMock(),
+        ) as from_url_mock:
+            await source.client.ping()
+            from_url_mock.assert_called_once()
```

Comment:
Not quite sure what it's trying to test?

`source.client.ping()` will delegate to `self._redis_client`, which is an `AsyncMock` instance. you probably should assert the `_redis_client` is called/awaited once.

Reply:
I just want to test ping here when_mutual_ssl_enabled without mocking `store_ssl_key` so that these lines also get cover with the tests.